### DISCUSSION TITLE
turbo run --dry=json says cache.local=false yet when I run the task, it says cache hit

### DISCUSSION THREAD

**Discussion by BenjaminVanRyseghem** [vercel/turborepo#9038] [Help] [answered]
### Summary

Hi,

I'm trying to add a CI task to know if I can skip the `yarn install` step that is quite slow.

To do so, I wanted to run `turbo run build --dry=json` to check the cache status, and thus decide if I need to proceed or if I can skip the following steps.

The issue I'm facing is that the output says that the local cache is missing, yet when I run the task "for real" it says that there is a cache hit.

The exact command run:

```
npx -y turbo@$TURBO_VERSION run "build" --dry=json | jq '.tasks[] | select(.command!="<NONEXISTENT>") | .cache.local'
```

(we need to filter out the task for workspaces with no `build` task)

The output looks like

```json
{
  "taskId": "PACKAGE#build",
  "task": "build",
  "package": "PACKAGE",
  "hash": "8e377e0ab82776c1",
  "inputs": {
    // lot of files
  },
  "hashOfExternalDependencies": "0d18ec9859b9392d",
  "cache": {
    "local": false,
    "remote": false,
    "status": "MISS",
    "timeSaved": 0
  },
  "command": "next build",
  "cliArguments": [],
  "outputs": null,
  "excludedOutputs": null,
  "logFile": "path/to/.turbo/turbo-build.log",
  "directory": "path/to",
  "dependencies": [
    "PACKAGE#transit"
  ],
  "dependents": [],
  "resolvedTaskDefinition": {
    "outputs": [],
    "cache": true,
    "dependsOn": [
      "transit"
    ],
    "inputs": [],
    "outputLogs": "full",
    "persistent": false,
    "env": [],
    "passThroughEnv": null,
    "interactive": false
  },
  "expandedOutputs": [],
  "framework": "nextjs",
  "envMode": "strict",
  "environmentVariables": {
    "specified": {
      "env": [],
      "passThroughEnv": null
    },
    "configured": [],
    "inferred": [],
    "passthrough": null
  }
}
```

when running the real task, the command is

```
turbo build
```

and the output starts with

```
PACKAGE:build: cache hit, replaying logs 74f0d18029cb48a5
```

So I'm quite puzzled, and a bit stuck on what I could do. Any pointer?

**Comment by anthonyshew** [comment, marked as answer]
Hm, that sounds more like a bug. Could you file an Issue with a reproduction? Thanks!

**Comment by BenjaminVanRyseghem** [comment]
This seems to only happen in AzureCI, so I don't really know how I could share that with you.

I'll see if I can have a reproduction....
